lib/Auth/Process/DatabaseCommand.php

```diff
@@ -23,10 +23,43 @@ public function __construct()
         $this->dbDriver = $this->databaseConnector->getDbDriver();
         $this->statisticsTableName = $this->databaseConnector->getStatisticsTableName();
         $this->detailedStatisticsTableName = $this->databaseConnector->getDetailedStatisticsTableName();
+        $this->ipStatisticsTableName = $this->databaseConnector->getIpStatisticsTableName();
         $this->identityProvidersMapTableName = $this->databaseConnector->getIdentityProvidersMapTableName();
         $this->serviceProvidersMapTableName = $this->databaseConnector->getServiceProvidersMapTableName();
     }
+    private function insertIpLogins($userId, $ip, $year, $month,$day, $time, $sourceIdp, $service)
+    {
+        $params = [
+          'userid' => $userId,
+          'ip' => $ip,
+          'sourceIdp' => $sourceIdp,
+          'service' => $service,
+          'year' => $year,
+          'month' => $month,
+          'day' => $day,
+          'time' => $time,
+        ];
+        $table = $this->ipStatisticsTableName;
+        $fields = array_keys($params);
```

Comment:
the original name `userid` column in mysql database is `user`. But, in PostgreSQL the word `user` is reserved.
So, you need to make the following check to support both, mysql and psql
```
        if ($user && $this->databaseConnector->getDetailedDays() > 0) {
            // write also into aggregated statistics
            self::writeLogin($year, $month, $day, $sourceIdp, $service);
            if ($this->dbDriver == 'pgsql') {
                // the word "user" is reserved from PostgreSQL
                $params['userid'] = $user;
            } else {
                $params['user'] = $user;
            }
            $table = $this->detailedStatisticsTableName;
        }
```